api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java

```diff
@@ -65,4 +77,131 @@ public void saveOrderGroup_shouldSaveOrderGroup() {
+		OrderGroup orderGroup = new OrderGroup();
+		
+		Encounter existingEncounter1 = encounterService.getEncounter(3);
+		Encounter existingEncounter2 = encounterService.getEncounter(4);
+
+		orderGroup.setEncounter(existingEncounter1);
+
+		assertEquals(0, dao.getOrderGroupsByEncounter(existingEncounter2).size());
+		assertEquals(0, dao.getOrderGroupsByEncounter(existingEncounter1).size());
+		
+		dao.saveOrderGroup(orderGroup);
+
+		assertEquals(0, dao.getOrderGroupsByEncounter(existingEncounter2).size());
+		assertEquals(1, dao.getOrderGroupsByEncounter(existingEncounter1).size());
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test(expected = APIException.class)
+	public void getOrderGroupsByEncounter_shouldHaveNullEncounter() {
+		OrderGroup orderGroup = new OrderGroup();
+		Encounter nullEncounter = null;
+
+		orderGroup.setEncounter(nullEncounter);
+		dao.getOrderGroupsByEncounter(nullEncounter);
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByPatient(Patient)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByPatient_shouldHaveMultipleOrderGroups() {
+		OrderGroup firstOrderGroup = new OrderGroup();
+		OrderGroup secondOrderGroup = new OrderGroup();
+		OrderGroup thirdOrderGroup = new OrderGroup();
+		
+		Patient existingPatient1 = patientService.getPatient(7);
+		Patient existingPatient2 = patientService.getPatient(8);
+
+		firstOrderGroup.setPatient(existingPatient1);
+		secondOrderGroup.setPatient(existingPatient1);
+		thirdOrderGroup.setPatient(existingPatient2);
+
+		assertEquals(0, dao.getOrderGroupsByPatient(existingPatient1).size());
+		assertEquals(0, dao.getOrderGroupsByPatient(existingPatient2).size());
+
+		dao.saveOrderGroup(firstOrderGroup);
+		dao.saveOrderGroup(secondOrderGroup);
+		dao.saveOrderGroup(thirdOrderGroup);
+
+		assertEquals(2, dao.getOrderGroupsByPatient(existingPatient1).size());
+		assertEquals(1, dao.getOrderGroupsByPatient(existingPatient2).size());
+		
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByPatient(Patient)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByPatient_shouldHaveNoOrderGroups() {
+		OrderGroup orderGroup = new OrderGroup();
+
+		Patient existingPatient1 = patientService.getPatient(7);
+		Patient existingPatient2 = patientService.getPatient(8);
+
+		orderGroup.setPatient(existingPatient1);
+
+		assertEquals(0, dao.getOrderGroupsByPatient(existingPatient2).size());
+		assertEquals(0, dao.getOrderGroupsByPatient(existingPatient1).size());
+		
+		dao.saveOrderGroup(orderGroup);
+
+		assertEquals(0, dao.getOrderGroupsByPatient(existingPatient2).size());
+		assertEquals(1, dao.getOrderGroupsByPatient(existingPatient1).size());
+	}
+	
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByPatient(Patient)}
+	 * @throws Exception
+	 */
+	@Test(expected = APIException.class)
+	public void getOrderGroupsByPatient_shouldHaveNullPatient() {
```

Comment:
Your test names are confusing. For instance, when you say that getOrderGroupsByPatient_shouldHaveNullPatient, does that mean all order groups should have a null patient? The same problem applies to your earlier tests.

Reply:
Already. I updated the names. However lmk if you have suggestions for better function names.